db/schema.rb

```diff
@@ -417,18 +419,20 @@
     t.integer "position"
     t.integer "question_set_id"
     t.datetime "updated_at", null: false
+    t.index "question_set_id, ((parent_id IS NULL))", name: "index_questions_on_question_set_id_parent_id_IS_NULL", unique: true, where: "(parent_id IS NULL)", comment: "Ensures max one root per question set"
     t.index ["question_set_id"], name: "index_questions_on_question_set_id"
   end
 
   create_table "response_sets", id: :serial, force: :cascade do |t|
     t.datetime "created_at", null: false
-    t.json "custom_data"
-    t.string "kind"
+    t.jsonb "custom_data"
     t.integer "loan_id", null: false
     t.integer "lock_version", default: 0, null: false
+    t.bigint "question_set_id", null: false
     t.datetime "updated_at", null: false
     t.integer "updater_id"
-    t.index ["loan_id", "kind"], name: "index_response_sets_on_loan_id_and_kind", unique: true
+    t.index ["loan_id", "question_set_id"], name: "index_response_sets_on_loan_id_and_question_set_id", unique: true
+    t.index ["question_set_id"], name: "index_response_sets_on_question_set_id"
```

Comment:
very very happy to have response sets belong to loans and qsets, and not rely on 'kind' anymore. thank you!